torch/csrc/cuda/Event.cpp

```diff
@@ -92,7 +92,7 @@ static PyObject * THCPEvent_get_cuda_event(THCPEvent *self) {
 
 static PyObject * THCPEvent_get_device(THCPEvent *self) {
   HANDLE_TH_ERRORS
-  return THPUtils_packInt64(self->cuda_event.device_index());
+  return THPDevice_New(self->cuda_event.device());
```

Comment:
(There also needs to be a test for that case)

Reply:
Right! What is the preferred behavior? Return -1 or raise an error?

Reply:
It should return `None`. The C++ case also needs to handle this. I think it will currently throw an exception. Instead Event::device() should return an `optional<Device>`.